pkg/controller/connectionsecret/connectionsecrets.go

```diff
@@ -56,6 +56,28 @@ func CreateOrUpdateConnectionSecrets(ctx *workflow.Context, k8sClient client.Cli
 		}
 	}
 
+	// same for serverless deployments
+	serverlessDeployments, _, err := ctx.Client.ServerlessInstances.List(context.Background(), project.ID(), &mongodbatlas.ListOptions{})
+	if err != nil {
+		return workflow.Terminate(workflow.DatabaseUserConnectionSecretsNotCreated, err.Error())
+	}
+	for _, c := range serverlessDeployments.Results {
+		found := false
+		for _, regularDeployment := range deployments {
+			if regularDeployment.Name == c.Name {
+				found = true
+				break
+			}
+		}
+
+		if !found {
```

Comment:
if `not found` ?

Reply:
Sure. New code handle serverless deployments that were not fetched thought v1(regular) and v1.5 API (advanced).

Reply:
I see :)